Tests/UnitTests/Purchasing/Purchases/PurchasesRestoreTests.swift

```diff
@@ -274,3 +274,78 @@ class PurchasesRestoreNoSetupTests: BasePurchasesTests {
     }
 
 }
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+class PurchasesRestoreTrackingTests: BasePurchasesTests {
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        self.setupPurchases()
+    }
+
+    func getMockDiagnosticsTracker() -> MockDiagnosticsTracker {
+        // swiftlint:disable:next force_cast
+        return self.diagnosticsTracker as! MockDiagnosticsTracker
+    }
```

Comment:
Not a big deal, but to avoid force casting I'm doing
```suggestion
    private func getMockDiagnosticsTracker() throws -> MockDiagnosticsTracker {
        return try XCTUnwrap(self.mockDiagnosticsTracker as? MockDiagnosticsTracker)
    }
```
Although you have to call it with `try`.
This way a failure is more friendly than a crash

Reply:
That makes a lot of sense. Thank you for the suggestion! 🙌

Reply:
Done in https://github.com/RevenueCat/purchases-ios/pull/4887/commits/993aeee732ec7c333b55fb3c818b83e45ea32ee0